include/boost/compute/types/fundamental.hpp

```diff
@@ -53,6 +53,13 @@ class vector_type
 
     vector_type()
     {
+
+    }
+
+    vector_type(const Scalar scalar)
```

Comment:
We should probably mark this `explicit` to prevent unintentional conversions from scalars to vectors.